app/routes/projects/upload_thumbnail.py

```
@@ -41,13 +43,75 @@ def upload_project_thumbnail(
     project.thumbnail = f"/api/projects/{project_id}/thumbnail"
     db.commit()
     db.refresh(project)
+    
+    # Invalidate cache after thumbnail upload
+    redis_service.delete(CACHE_KEYS["PROJECTS_LIST"])
 
     return {"url": project.thumbnail, "project": {"id": project.id, "thumbnail": project.thumbnail}}
 
 @router.get("/{project_id}/thumbnail", summary="Get project thumbnail image")
-def get_project_thumbnail(project_id: int, db: Session = Depends(get_db)):
+def get_project_thumbnail(project_id: int, request: Request, db: Session = Depends(get_db)):
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)